src/commands/topics.ts

```diff
@@ -166,6 +167,66 @@ async function editTopicConfig(topic: KafkaTopic): Promise<void> {
   editConfigForm.onDidDispose(() => disposable.dispose());
 }
 
+async function produceMessageFromFile(topic: KafkaTopic) {
+  const options: vscode.OpenDialogOptions = {
+    canSelectMany: false,
+    openLabel: "Select JSON file",
+    filters: {
+      "JSON files": ["json"],
+    },
+  };
+
+  const fileUri = await vscode.window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    const filePath = fileUri[0].fsPath;
+    const message = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+
+    if (!topic) {
+      vscode.window.showErrorMessage("No topic selected.");
+      return;
+    }
+
+    if (!message.key || !message.value) {
+      vscode.window.showErrorMessage(`Message must have a key and a value.`);
+    }
+
+    const sidecar = await getSidecar();
+    const clusterId = topic.clusterId;
+    const connectionId = topic.connectionId;
+    const topicName = topic.name;
+
+    const recordsApi = sidecar.getRecordsV3Api(clusterId, connectionId);
+
+    const type = topic.connectionType === "LOCAL" ? undefined : "JSON";
+
+    const msgKeyAndVal = {
+      key: { data: message.key, ...(type && { type }) },
+      value: { data: message.value, ...(type && { type }) },
+    };
+
+    try {
+      let response = await recordsApi.produceRecord({
+        topic_name: topicName,
+        cluster_id: clusterId,
+        ProduceRequest: {
+          headers: message.headers.map((header: any) => ({
+            name: header.key ? header.key : header.name,
+            value: header.value,
+          })),
+          key: msgKeyAndVal.key,
+          value: msgKeyAndVal.value,
+        },
+      });
+
+      if (response) {
+        vscode.window.showInformationMessage(`Success: Produced message to topic ${topic.name}.`);
+      }
+    } catch (error: any) {
+      vscode.window.showErrorMessage(`Failed to produce message: ${error.message}`);
+    }
```

Comment:
We can safely remove this `if (response)` since we'll either get a successful response, or the client code will raise a `ResponseError` for anything with a `status` >= 400. For the latter, we could use [logResponseError()](https://github.com/confluentinc/vscode/blob/main/src/errors.ts#L26) (just for the output channel) and handle awaiting the response text for the notification, something like this:
```suggestion
      vscode.window.showInformationMessage(`Success: Produced message to topic ${topic.name}.`);
    } catch (error: any) {
      logResponseError(error, "topic produce from file"); // not sending to Sentry by default
      if (error instanceof ResponseError) {
        const body = await error.response.clone().text();
        vscode.window.showErrorMessage(`Error response while trying to produce message: ${error.response.status} ${error.response.statusText}: ${body}`);
      } else {
        vscode.window.showErrorMessage(`Failed to produce message: ${error.message}`);
      }
    }
```